mteb/models/openai_models.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import logging
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+
+from mteb.import_utils import requires_package
+from mteb.model_meta import ModelMeta
+from mteb.models.utils import corpus_to_texts
+
+logger = logging.getLogger(__name__)
+
+
+class OpenAIWrapper:
+    def __init__(self, model_name: str, embed_dim: int | None = None) -> None:
+        requires_package(self, "openai", "Openai text embedding")
+        from openai import OpenAI
+
+        self._client = OpenAI()
+        self._model_name = model_name
+        self._embed_dim = embed_dim
+
+    def encode(self, sentences: list[str], **kwargs: Any) -> torch.Tensor | np.ndarray:
```

Comment:
I tested the voyage model on several datasets, one of which has very long texts. The scores are in the results/ folder. I also added a rpm and tpm limit for the voyage model. I'm not sure if requests will often go over the limits as they are very large, but it will definitely protect against errors.
For the openai model I don't have an account with free tokens, so I was able to test. Perhaps you can give me a token or test on your own?

Reply:
Sounds good. I will merge it in and then run the test tomorrow (if there are any issues, it is probably easier for the person running it to fix them regardless).